services-core/src/gas_price.rs

```diff
@@ -23,14 +25,48 @@ pub trait GasPriceEstimating: Send + Sync {
     async fn estimate_with_limits(&self, gas_limit: f64, time_limit: Duration) -> Result<f64>;
 }
 
+#[async_trait::async_trait]
+pub trait Transport: Send + Sync {
+    async fn get_json<T: DeserializeOwned>(&self, url: &'static str) -> Result<T>;
+}
+
+#[async_trait::async_trait]
+impl Transport for HttpClient {
+    async fn get_json<T: DeserializeOwned>(&self, url: &'static str) -> Result<T> {
+        let uri = Uri::from_static(url);
```

Comment:
This is problematic because `from_static` panics if the url is invalid.Should use whatever the normal contructor is that cn return an error.